common/cards/alter-egos/effects/turtle-shell.ts

```diff
@@ -19,35 +53,158 @@ const TurtleShell: Attach = {
 		attach.attachCondition,
 		query.not(query.slot.active),
 	),
+	onCreate(game: GameModel, component: CardComponent) {
+		if (game.id in lastActiveHermit.values) return
+		lastActiveHermit.set(game, {})
+
+		const newObserver = game.components.new(ObserverComponent, component.entity)
+
+		game.components.filter(PlayerComponent).forEach((player) => {
+			newObserver.subscribe(
+				player.hooks.onActiveRowChange,
+				(oldHermit, newHermit) => {
+					if (oldHermit !== null) {
+						if (
+							lastActiveHermit.get(game)[player.entity]?.hermit.entity ===
+							newHermit.entity
+						) {
+							// Ladder was used to move active hermit to new row
+							return
+						}
+					}
+					updateRecord(game, player.entity, {
+						hermit: newHermit,
+						stage:
+							game.currentPlayerEntity === player.entity
+								? 'player-activated'
+								: 'opponent-activated',
+					})
+				},
+			)
+
+			newObserver.subscribe(player.hooks.onTurnStart, () => {
+				game.components.filter(PlayerComponent).forEach((player) => {
+					const activeHermit = player.getActiveHermit()
+					if (activeHermit === null) {
+						delete lastActiveHermit.get(game)[player.entity]
+						return
+					}
+
+					const activeInfo = lastActiveHermit.get(game)[player.entity]
+					assert(activeInfo)
+					switch (activeInfo.stage) {
+						case 'opponent-activated':
+							updateRecord(game, player.entity, {
+								hermit: activeHermit,
+								stage: 'player-activated',
+							})
+							break
+						case 'player-activated':
+							updateRecord(game, player.entity, {
+								hermit: activeHermit,
+								stage: 'defend-then-discard',
+							})
+							break
+						case 'defend-then-discard':
+							updateRecord(game, player.entity, {
+								hermit: activeHermit,
+								stage: 'too-late',
+							})
+							break
+					}
+				})
+			})
+		})
+	},
 	onAttach(
 		game: GameModel,
 		component: CardComponent,
 		observer: ObserverComponent,
 	) {
 		const {player} = component
 		let activated = false
+		let shouldDiscard = false
+
+		if (query.slot.active(game, component.slot)) {
+			// Only possible with Emerald or Grian's Borrow
```

Comment:
It would be better to give a more generic explanation here. It is too likely for comments that mention something specific to  become wrong in the future and confuse someone.
```
// Only possible with cards that move attachables. For example, Emerald or Grian's Borrow.
``